### PR TITLE
Fix for handling multiple records in ballerina to openapi command

### LINKED ISSUE #143: Client code generator not up-to-date with latest HTTP changes
It seems the client code generator hasn't been updated to align with the latest changes in the HTTP module. For example, the following line is from a generated client:
```ballerina
var response = listPetsEp->get("/pets", message = request);
```

### LINKED ISSUE #148: [Bal -> OpenAPI] Generator doesn't generate more than one record
**Description:**
When the bal file has more than one record, open api generator generates only one record in the yaml/json file. 

**Steps to reproduce:**
```ballerina
import ballerina/http;

listener http:Listener personEp = new (9090);

type Person record {
    int id;
    string fname;
    string lname;
    int age;
};

type PersonInput record {
    string fname;
    string lname;
    int age;
};

Person[] personList = [];

service http:Service /api/person on personEp {

    resource function post .(@http:Payload {} PersonInput person) returns json {

        Person newPerson = {
            id: personList.length(),
            fname: person.fname,
            lname: person.lname,
            age: person.age
        };

        personList.push(newPerson);
        json|error greeting = newPerson.cloneWithType(json);
        if (greeting is json) {
            return greeting;
        }
    }

    resource function put .(@http:Payload {} Person person) returns json {
        int index = getIndexOfPerson(person.id);
        personList[index] = person;
        json result = {"Message": "Entry updated"};
        return result;
    }
}

function getIndexOfPerson(int id) returns int {
    int index = -1;
    foreach int i in 0 ... personList.length() {
        Person entry = personList[i];
        int entryId = entry.id;
        if (entryId == id) {
            return entryId;
        }
    }
    return index;
}
```

This code generates only one record in the yaml 
```yaml
components:
  schemas:
    Person:
      type: object
      properties:
        fname:
          type: string
        lname:
          type: string
        id:
          type: integer
          format: int32
        age:
          type: integer
          format: int32
```
**Affected Product Version:**
slalpha3 

**OS, DB, other environment details and versions:**    
Ubuntu 20.04

### PR DESCRIPTION
**Pull request by lnash94**
Fix #143 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java [hunk 1]
```diff
@@ -699,21 +698,26 @@ private void handleRecordPayload(SimpleNameReferenceNode queryParam, Map<String,
...
+                        schema = components.getSchemas();
                     }
                     schemaProperties.put(field.getKey(), property);
                 }
                 componentSchema.setProperties(schemaProperties);
             }
         }
-
-        if (!schema.containsKey(componentName)) {
+        if ((schema != null) && (!schema.containsKey(componentName))) {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java:713]:
```suggestion
        if (schema != null && !schema.containsKey(componentName)) {
```

File: openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+};
+service /payloadV on helloEp {
+    resource function post pets(@http:Payload{} Pet payload) returns  http:Ok {
+
+    }
+    resource function put pets(@http:Payload{} Tag payload) returns  http:Ok {
+
+    }
+}
+
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal:25]: Shall we fix formatting in here?

File: openapi-cli/src/test/java/org/ballerinalang/ballerina/OpenApiConverterUtilsTest.java [hunk 3]
```diff
@@ -375,6 +375,12 @@ public void testListners03() throws OpenApiConverterException {
         compareWithGeneratedFile(ballerinaFilePath, "listener_scenario03.yaml");
     }
 
+    @Test(description = "Generate OpenAPI spec for multiple listners")
+    public void tetsMultipleRecords() throws OpenApiConverterException {
```
Comment (suganyasuven) [inline openapi-cli/src/test/java/org/ballerinalang/ballerina/OpenApiConverterUtilsTest.java:379]:
```suggestion
    public void testMultipleRecords() throws OpenApiConverterException {
```

File: openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import ballerina/http;
+
+listener http:Listener helloEp = new (9090);
+type TagType record {
+    int id;
+    string[] category;
+};
```
Comment (suganyasuven) [inline openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal:7]: Shall we add a line between each record and remove the extra line at the end of the file?

File: openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal [hunk 5]
```diff
@@ -0,0 +1,28 @@
+importballerina/http;
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/resources/ballerina-to-openapi/rb_scenario11.bal:1]:
```suggestion
import ballerina/http;
```

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by suganyasuven — APPROVED** [review]

**Pull request merged**